src/main/java/BankingSystem.java

```diff
@@ -0,0 +1,209 @@
+import dao.AccountDao;
+import model.Account;
+import model.Card;
+import model.CardGenerator;
+
+import java.util.Scanner;
+
+public class BankingSystem {
+
+    private final AccountDao dao;
+    private final CardGenerator cardGenerator;
+    private final Scanner scanner;
+    private Account currentAccount;
+
+    public BankingSystem(AccountDao dao) {
+        this.dao = dao;
+        cardGenerator = new CardGenerator();
+        scanner = new Scanner(System.in);
+    }
+
+    public void showMenu() {
+        while (true) {
+            welcomeMenu();
+
+            String input = scanner.next();
+
+            switch (input) {
+                case "0" -> {
+                    scanner.close();
+                    System.exit(0);
+                }
+                case "1" -> registerAccount();
+                case "2" -> logIntoAccount();
+                default -> System.out.println("You've entered invalid menu item.\n");
+            }
+        }
+    }
+
+    private void welcomeMenu() {
+        System.out.println("""
+                1. Create an account
+                2. Log into account
+                0. Exit""");
+    }
+
+    private void registerAccount() {
+        Card card = cardGenerator.generateCredentials();
+        dao.save(new Account(card, 0));
+        System.out.printf("""
+                Your card has been created
+                Your card number:
+                %s
+                Your card PIN:
+                %s""", card.number(), card.pin());
+
+        System.out.print("\n".repeat(2));
+    }
+
+    private void logIntoAccount() {
+        System.out.println("\nEnter your card number:");
+        String inCardNum = scanner.next();
+        System.out.println("Enter your PIN:");
+        String inCardPIN = scanner.next();
+        boolean isValid = accountValidation(inCardNum, inCardPIN);
+
+        if (isValid) {
+            System.out.println("\nYou have successfully logged in!\n");
+            dao.get(inCardNum, inCardPIN).ifPresentOrElse(account -> currentAccount = account,
+                    () -> System.out.println("\nWrong card number or PIN!\n"));
+            accountMenu();
+        }
+    }
+
+    private boolean accountValidation(String inputtedCardNumber, String inputtedCardPIN) {
+        return dao.get(inputtedCardNumber, inputtedCardPIN).isPresent();
+    }
```

Comment:
You duplicate call to the `dao.get`. Can you remove this method?

Reply:
Removed